python/tank/commands/push_pc.py

```diff
@@ -103,30 +129,99 @@ def run_interactive(self, log, args):
             log.info(" - [%d] %s (%s)" % (pc["id"], pc["code"], local_path))
         log.info("")
         
-        answer = raw_input("Please type in the id of the configuration to push to (ENTER to exit): " )
+        answer = raw_input(
+            "Please type in the id of the configuration to push to (ENTER to exit): "
+        )
         if answer == "":
             raise TankError("Aborted by user.")
         try:
             target_pc_id = int(answer)
         except:
             raise TankError("Please enter a number!")
-        
-        if target_pc_id not in [ x["id"] for x in pipeline_configs]:
-            raise TankError("Id was not found in the list!")
 
-        target_pc_path = path_hash[target_pc_id]
+        self._run(
+            log,
+            **(self._validate_parameters({
+                "target_id": target_pc_id,
+                "use_symlink": use_symlink,
+            }))
+        )
+
+    def _preflight(self):
+        """
+        Performs actions needed in both interactive/non interactive modes.
+
+        Validate we can run a push in the current context.
+
+        :raises: TankError if pushing is invalid.
+        """
+        # get list of all PCs for this project
+        if self.tk.pipeline_configuration.is_site_configuration():
+            raise TankError("You can't push the site configuration.")
+
+        if self.tk.pipeline_configuration.is_unmanaged():
+            raise TankError("You can't push an unmanaged configuration.")
+
+        project_id = self.tk.pipeline_configuration.get_project_id()
+
+        self._pipeline_configs = self.tk.shotgun.find(
+            constants.PIPELINE_CONFIGURATION_ENTITY,
+            [["project", "is", {"type": "Project", "id": project_id}]],
+            ["code", "linux_path", "windows_path", "mac_path"]
+        )
+
+        # We should have at least one pipeline config (the current one)
+        # We need a second one to push to, obviously...
+        if len(self._pipeline_configs) == 1:
```

Comment:
I had exactly the same thought, but I think there is always at least one pipeline config: the current one. I changed the code anyway as we both thought the same thing, even if `== 1` is correct.